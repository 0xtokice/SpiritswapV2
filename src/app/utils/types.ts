/**
 * app/utils/types.ts
 *
 * Describe common types, interfaces, and enums
 *
 */

export interface Wallet {
  portfolio: { [key in keyof typeof Token]?: number | undefined };
  rates: { [key in keyof typeof Token]?: number | undefined };
}

export interface TokenValue {
  token: string | Token;
  value: number;
}

export interface ChartProps {
  pairAddress: string | undefined;
  outTokenAddress: string;
  inTokenAddress: string;
  currency: string;
}

export interface TokenCurrencyConversionRate {
  token: Token;
  currency: Currency;
  conversionRate: number;
}

export interface CurrencyOption {
  value: string;
  id: number;
  type: string;
}

export type TokenCurrencyConversionRateMap = {
  [key in keyof typeof Token]?: TokenCurrencyConversionRate | undefined;
};

export enum Sign {
  POSITIVE,
  NEGATIVE,
  NEUTRAL,
}

export enum Currency {
  EUR = 'EUR',
  USD = 'USD',
}

export enum AmountTypes {
  BALANCE = 'balance',
  STAKED = 'staked',
}

export enum Token {
  FTM = 'FTM',
  SPIRIT = 'SPIRIT',
  PAC = 'PAC',
  BTC0x = '0XBTC',
  GIVE2 = '2GIVE',
  ABT = 'ABT',
  ACT = 'ACT',
  ACTN = 'ACTN',
  ADA = 'ADA',
  ADD = 'ADD',
  ADX = 'ADX',
  AE = 'AE',
  AEON = 'AEON',
  AEUR = 'AEUR',
  AGI = 'AGI',
  AGRS = 'AGRS',
  AION = 'AION',
  ALGO = 'ALGO',
  AMB = 'AMB',
  AMP = 'AMP',
  AMPL = 'AMPL',
  ANT = 'ANT',
  APPC = 'APPC',
  ARDR = 'ARDR',
  ARG = 'ARG',
  ARK = 'ARK',
  ARN = 'ARN',
  ARNX = 'ARNX',
  ARY = 'ARY',
  AST = 'AST',
  ATM = 'ATM',
  ATOM = 'ATOM',
  AUDR = 'AUDR',
  AUTO = 'AUTO',
  AYWA = 'AYWA',
  BAB = 'BAB',
  BAL = 'BAL',
  BAND = 'BAND',
  BAT = 'BAT',
  BAY = 'BAY',
  BCBC = 'BCBC',
  BCC = 'BCC',
  BCD = 'BCD',
  BCH = 'BCH',
  BCIO = 'BCIO',
  BCN = 'BCN',
  BCO = 'BCO',
  BCPT = 'BCPT',
  BDL = 'BDL',
  BEAM = 'BEAM',
  BELA = 'BELA',
  BIX = 'BIX',
  BLCN = 'BLCN',
  BLK = 'BLK',
  BLOCK = 'BLOCK',
  BLZ = 'BLZ',
  BNB = 'BNB',
  BNT = 'BNT',
  BNTY = 'BNTY',
  BOOTY = 'BOOTY',
  BOS = 'BOS',
  BPT = 'BPT',
  BQ = 'BQ',
  BRD = 'BRD',
  BSD = 'BSD',
  BSV = 'BSV',
  BTC = 'BTC',
  BTCD = 'BTCD',
  BTCH = 'BTCH',
  BTCP = 'BTCP',
  BTCZ = 'BTCZ',
  BTDX = 'BTDX',
  BTG = 'BTG',
  BTM = 'BTM',
  BTS = 'BTS',
  BTT = 'BTT',
  BTX = 'BTX',
  BURST = 'BURST',
  BZE = 'BZE',
  CALL = 'CALL',
  CC = 'CC',
  CDN = 'CDN',
  CDT = 'CDT',
  CHAIN = 'CHAIN',
  CHAT = 'CHAT',
  CHIPS = 'CHIPS',
  CIX = 'CIX',
  CLAM = 'CLAM',
  CLOAK = 'CLOAK',
  CMM = 'CMM',
  CMT = 'CMT',
  CND = 'CND',
  CNX = 'CNX',
  CNY = 'CNY',
  COB = 'COB',
  COLX = 'COLX',
  COMP = 'COMP',
  COQUI = 'COQUI',
  CRED = 'CRED',
  CRPT = 'CRPT',
  CRW = 'CRW',
  CS = 'CS',
  CTR = 'CTR',
  CTXC = 'CTXC',
  CVC = 'CVC',
  D = 'D',
  DAI = 'DAI',
  DASH = 'DASH',
  DAT = 'DAT',
  DATA = 'DATA',
  DBC = 'DBC',
  DCN = 'DCN',
  DCR = 'DCR',
  DEEZ = 'DEEZ',
  DENT = 'DENT',
  DEW = 'DEW',
  DGB = 'DGB',
  DGD = 'DGD',
  DLT = 'DLT',
  DNT = 'DNT',
  DOCK = 'DOCK',
  DOGE = 'DOGE',
  DOT = 'DOT',
  DRGN = 'DRGN',
  DROP = 'DROP',
  DTA = 'DTA',
  DTH = 'DTH',
  DTR = 'DTR',
  EBST = 'EBST',
  ECA = 'ECA',
  EDG = 'EDG',
  EDO = 'EDO',
  EDOGE = 'EDOGE',
  ELA = 'ELA',
  ELEC = 'ELEC',
  ELF = 'ELF',
  ELIX = 'ELIX',
  ELLA = 'ELLA',
  EMC = 'EMC',
  EMC2 = 'EMC2',
  ENG = 'ENG',
  ENJ = 'ENJ',
  ENTRP = 'ENTRP',
  EON = 'EON',
  EOP = 'EOP',
  EOS = 'EOS',
  EQLI = 'EQLI',
  EQUA = 'EQUA',
  ETC = 'ETC',
  ETH = 'ETH',
  ETHOS = 'ETHOS',
  ETN = 'ETN',
  ETP = 'ETP',
  EUR = 'EUR',
  EVX = 'EVX',
  EXMO = 'EXMO',
  EXP = 'EXP',
  FAIR = 'FAIR',
  FCT = 'FCT',
  FIL = 'FIL',
  FJC = 'FJC',
  FLDC = 'FLDC',
  FLO = 'FLO',
  FSN = 'FSN',
  FTC = 'FTC',
  FUEL = 'FUEL',
  FUN = 'FUN',
  GAME = 'GAME',
  GAS = 'GAS',
  GBP = 'GBP',
  GBX = 'GBX',
  GBYTE = 'GBYTE',
  GENERIC = 'GENERIC',
  GIN = 'GIN',
  GLXT = 'GLXT',
  GMR = 'GMR',
  GNO = 'GNO',
  GNT = 'GNT',
  GOLD = 'GOLD',
  GRC = 'GRC',
  GRIN = 'GRIN',
  GRS = 'GRS',
  GSC = 'GSC',
  GTO = 'GTO',
  GUP = 'GUP',
  GUSD = 'GUSD',
  GVT = 'GVT',
  GXS = 'GXS',
  GZR = 'GZR',
  HIGHT = 'HIGHT',
  HODL = 'HODL',
  HOT = 'HOT',
  HPB = 'HPB',
  HSR = 'HSR',
  HT = 'HT',
  HTML = 'HTML',
  HUC = 'HUC',
  HUSH = 'HUSH',
  ICN = 'ICN',
  ICX = 'ICX',
  IGNIS = 'IGNIS',
  ILK = 'ILK',
  INK = 'INK',
  INS = 'INS',
  ION = 'ION',
  IOP = 'IOP',
  IOST = 'IOST',
  IOTX = 'IOTX',
  IQ = 'IQ',
  ITC = 'ITC',
  JNT = 'JNT',
  JPY = 'JPY',
  KCS = 'KCS',
  KIN = 'KIN',
  KLOWN = 'KLOWN',
  KMD = 'KMD',
  KNC = 'KNC',
  KRB = 'KRB',
  LBC = 'LBC',
  LEND = 'LEND',
  LEO = 'LEO',
  LINK = 'LINK',
  LKK = 'LKK',
  LOOM = 'LOOM',
  LPT = 'LPT',
  LRC = 'LRC',
  LSK = 'LSK',
  LTC = 'LTC',
  LUN = 'LUN',
  MAID = 'MAID',
  MANA = 'MANA',
  MATIC = 'MATIC',
  MCAP = 'MCAP',
  MCO = 'MCO',
  MDA = 'MDA',
  MDS = 'MDS',
  MED = 'MED',
  MEETONE = 'MEETONE',
  MFT = 'MFT',
  MIOTA = 'MIOTA',
  MITH = 'MITH',
  MKR = 'MKR',
  MLN = 'MLN',
  MNX = 'MNX',
  MNZ = 'MNZ',
  MOAC = 'MOAC',
  MOD = 'MOD',
  MONA = 'MONA',
  MSR = 'MSR',
  MTH = 'MTH',
  MTL = 'MTL',
  MUSIC = 'MUSIC',
  MZC = 'MZC',
  NANO = 'NANO',
  NAS = 'NAS',
  NAV = 'NAV',
  NCASH = 'NCASH',
  NDZ = 'NDZ',
  NEBL = 'NEBL',
  NEO = 'NEO',
  NEOS = 'NEOS',
  NEU = 'NEU',
  NEXO = 'NEXO',
  NGC = 'NGC',
  NIO = 'NIO',
  NLC2 = 'NLC2',
  NLG = 'NLG',
  NMC = 'NMC',
  NMR = 'NMR',
  NPXS = 'NPXS',
  NULS = 'NULS',
  NXS = 'NXS',
  NXT = 'NXT',
  OAX = 'OAX',
  OK = 'OK',
  OMG = 'OMG',
  OMNI = 'OMNI',
  ONG = 'ONG',
  ONT = 'ONT',
  OOT = 'OOT',
  OST = 'OST',
  OX = 'OX',
  OXT = 'OXT',
  PART = 'PART',
  PASC = 'PASC',
  PASL = 'PASL',
  PAX = 'PAX',
  PAY = 'PAY',
  PAYX = 'PAYX',
  PINK = 'PINK',
  PIRL = 'PIRL',
  PIVX = 'PIVX',
  PLR = 'PLR',
  POA = 'POA',
  POE = 'POE',
  POLIS = 'POLIS',
  POLY = 'POLY',
  POT = 'POT',
  POWR = 'POWR',
  PPC = 'PPC',
  PPP = 'PPP',
  PPT = 'PPT',
  PRE = 'PRE',
  PRL = 'PRL',
  PUNGO = 'PUNGO',
  PURA = 'PURA',
  QASH = 'QASH',
  QIWI = 'QIWI',
  QLC = 'QLC',
  QRL = 'QRL',
  QSP = 'QSP',
  QTUM = 'QTUM',
  R = 'R',
  RADS = 'RADS',
  RAP = 'RAP',
  RCN = 'RCN',
  RDD = 'RDD',
  RDN = 'RDN',
  REN = 'REN',
  REP = 'REP',
  REPV2 = 'REPV2',
  REQ = 'REQ',
  RHOC = 'RHOC',
  RIC = 'RIC',
  RISE = 'RISE',
  RLC = 'RLC',
  RPX = 'RPX',
  RUB = 'RUB',
  RVN = 'RVN',
  RYO = 'RYO',
  SAFE = 'SAFE',
  SAI = 'SAI',
  SALT = 'SALT',
  SAN = 'SAN',
  SBD = 'SBD',
  SBERBANK = 'SBERBANK',
  SC = 'SC',
  SHIFT = 'SHIFT',
  SIB = 'SIB',
  SIN = 'SIN',
  SKY = 'SKY',
  SLR = 'SLR',
  SLS = 'SLS',
  SMART = 'SMART',
  SNGLS = 'SNGLS',
  SNM = 'SNM',
  SNT = 'SNT',
  SOC = 'SOC',
  SPANK = 'SPANK',
  SPHTX = 'SPHTX',
  SRN = 'SRN',
  STAK = 'STAK',
  START = 'START',
  STEEM = 'STEEM',
  STORJ = 'STORJ',
  STORM = 'STORM',
  STQ = 'STQ',
  STRAT = 'STRAT',
  STX = 'STX',
  SUB = 'SUB',
  SUMO = 'SUMO',
  SYS = 'SYS',
  TAAS = 'TAAS',
  TAU = 'TAU',
  TBX = 'TBX',
  TEL = 'TEL',
  TEN = 'TEN',
  TERN = 'TERN',
  TGCH = 'TGCH',
  THETA = 'THETA',
  TIX = 'TIX',
  TKN = 'TKN',
  TKS = 'TKS',
  TNB = 'TNB',
  TNC = 'TNC',
  TNT = 'TNT',
  TOMO = 'TOMO',
  TPAY = 'TPAY',
  TRIG = 'TRIG',
  TRTL = 'TRTL',
  TRX = 'TRX',
  TUSD = 'TUSD',
  TZC = 'TZC',
  UBQ = 'UBQ',
  UMA = 'UMA',
  UNI = 'UNI',
  UNITY = 'UNITY',
  USD = 'USD',
  USDC = 'USDC',
  USDT = 'USDT',
  UTK = 'UTK',
  VERI = 'VERI',
  VET = 'VET',
  VIA = 'VIA',
  VIB = 'VIB',
  VIBE = 'VIBE',
  VIVO = 'VIVO',
  VRC = 'VRC',
  VRSC = 'VRSC',
  VTC = 'VTC',
  VTHO = 'VTHO',
  WABI = 'WABI',
  WAN = 'WAN',
  WAVES = 'WAVES',
  WAX = 'WAX',
  WBTC = 'WBTC',
  WGR = 'WGR',
  WICC = 'WICC',
  WINGS = 'WINGS',
  WPR = 'WPR',
  WTC = 'WTC',
  X = 'X',
  XAS = 'XAS',
  XBC = 'XBC',
  XBP = 'XBP',
  XBY = 'XBY',
  XCP = 'XCP',
  XDN = 'XDN',
  XEM = 'XEM',
  XIN = 'XIN',
  XLM = 'XLM',
  XMCC = 'XMCC',
  XMG = 'XMG',
  XMO = 'XMO',
  XMR = 'XMR',
  XMY = 'XMY',
  XP = 'XP',
  XPA = 'XPA',
  XPM = 'XPM',
  XRP = 'XRP',
  XSG = 'XSG',
  XTZ = 'XTZ',
  XUC = 'XUC',
  XVC = 'XVC',
  XVG = 'XVG',
  XZC = 'XZC',
  YFI = 'YFI',
  YOYOW = 'YOYOW',
  ZCL = 'ZCL',
  ZEC = 'ZEC',
  ZEL = 'ZEL',
  ZEN = 'ZEN',
  ZEST = 'ZEST',
  ZIL = 'ZIL',
  ZILLA = 'ZILLA',
  ZRX = 'ZRX',
}
