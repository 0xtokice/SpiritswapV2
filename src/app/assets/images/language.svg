<svg width="16" height="17" viewBox="0 0 16 17" fill="none" xmlns="http://www.w3.org/2000/svg">
	<path id="Layer" class="s0" d="m8 1c4.2 0 7.5 3.3 7.5 7.5c0 4.2-3.3 7.5-7.5 7.5" stroke="white" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round" />
	<path id="Layer" class="s0" d="m8 16c-4.2 0-7.5-3.3-7.5-7.5c0-4.2 3.3-7.5 7.5-7.5" stroke="white" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round" />
	<path id="Layer" class="s0" d="m9.5 1.9c2.4 4 2.4 9.2 0 13.2c-0.7 1.2-2.3 1.2-3 0c-2.4-4-2.4-9.2 0-13.2c0.7-1.2 2.3-1.2 3 0z" stroke="white" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round" />
	<path id="Layer" class="s0" d="m0.5 8.5h15" stroke="white" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round" />
</svg>