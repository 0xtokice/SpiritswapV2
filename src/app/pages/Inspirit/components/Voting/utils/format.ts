export const formatVotes = (base: number) =>
  `${`${(base / 1000000).toFixed(1)}`}M`;
