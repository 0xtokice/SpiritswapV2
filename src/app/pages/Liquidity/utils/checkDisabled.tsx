const checkDisabled = (firstToken, secondToken, balanceOne, balanceTwo) => {
  return (
    firstToken.value === '' ||
    firstToken.value === '0' ||
    firstToken.value === '0.0' ||
    secondToken.value === '' ||
    secondToken.value === '0' ||
    secondToken.value === '0.0' ||
    balanceOne === 0 ||
    balanceTwo === 0 ||
    isNaN(firstToken.value) ||
    isNaN(secondToken.value) ||
    isNaN(balanceOne) ||
    isNaN(balanceTwo)
  );
};

export default checkDisabled;
