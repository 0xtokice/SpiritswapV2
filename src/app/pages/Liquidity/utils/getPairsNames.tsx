const getPairsNames = (pairsList: string[]) => {
  const arrayLength = pairsList?.length;
  let finalString = '';
  const isTheLast = arrayLength - 1;
  pairsList?.forEach((item, index) => {
    if (isTheLast === index) return (finalString += item);

    finalString += item + ' + ';
  });
  return finalString;
};

export default getPairsNames;
