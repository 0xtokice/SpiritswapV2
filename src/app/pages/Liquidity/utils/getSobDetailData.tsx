import { tokenData } from 'utils/data/types';
import { LiquidityDetailProps } from './getDetailData';

const getSobDetailData = (pair: tokenData) => {
  let data: LiquidityDetailProps[] = [];

  const { lpSupply, amount, tokensAmounts } = pair;

  const poolTokenPercentage = lpSupply
    ? `${((parseFloat(amount) / lpSupply) * 100).toFixed(5)} %`
    : 'Unknown'; // get the share pool percentaje

  tokensAmounts?.forEach(tokenWAmount => {
    data.push({
      detailTitle: `Pooled ${tokenWAmount.token.symbol}:`,
      detailValue: tokenWAmount.amount ? `${tokenWAmount.amount}` : 'Unknown',
    });
  });

  data = [
    ...data,
    {
      detailTitle: 'Your pool share:',
      detailValue: `${poolTokenPercentage}`,
    },
  ];

  data = data.map((item, i) => {
    return { ...item, id: i };
  });

  return data;
};

export default getSobDetailData;
