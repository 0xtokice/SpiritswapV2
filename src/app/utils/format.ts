import BigNumber from 'bignumber.js';

// JavaScript numbers use exponential notation for positive exponents of 21 and above.
// JavaScript numbers use exponential notation for negative exponents of -7 and below. We need more.
BigNumber.config({ EXPONENTIAL_AT: [-21, 21] });
/**
 * Format token amount to at least 4 decimals.
 * @param amount amount to format.
 * @returns formatted amount.
 */
export const formatTokenAmount = (
  amount: string = '0',
  decimals: number = 0,
) => {
  let shiftedAmount = amount;
  if (decimals) {
    shiftedAmount = (Number(amount) / 10 ** decimals).toString();
  }
  const parsedAmount = parseFloat(shiftedAmount);
  if (parsedAmount === 0 || isNaN(Number(shiftedAmount))) {
    return '0';
  }

  let decimalPlaces = 3;
  const absAmount = Math.abs(parsedAmount);
  while (absAmount < 1 / 10 ** decimalPlaces) {
    decimalPlaces++;
  }

  return new BigNumber(
    parseFloat(new BigNumber(parsedAmount).toFixed(decimalPlaces + 1, 0)),
  ).toString();
};
