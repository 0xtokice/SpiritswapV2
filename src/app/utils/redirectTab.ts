export const openInNewTab = url => {
  const newWindow = window.open(url, 'noopener,noreferrer');
  if (newWindow) newWindow.opener = null;
};

export const openInSelfTab = url => {
  window.location.assign(url);
};

export const onClickUrl =
  (url: string | undefined): (() => void) =>
  () =>
    openInNewTab(url);
