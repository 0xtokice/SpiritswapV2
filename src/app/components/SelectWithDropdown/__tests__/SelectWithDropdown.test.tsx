import { render, fireEvent } from '@testing-library/react';
import { LanguageItems } from 'app/utils/constants';
import { SelectWithDropdown } from '../index';
import { ReactComponent as LanguageIcon } from 'app/assets/images/language.svg';
import { ThemeWrapper } from 'app/components/shared/testing/ThemeWrapper';
import ThemeProvider from 'theme';
import { ChakraThemeProvider } from 'theme/chakra';

describe('<SelectWithDropdown />', () => {
  const props = {
    items: LanguageItems,
    selectedId: 1,
    icon: <LanguageIcon />,
  };

  const renderComponent = props =>
    render(
      <ChakraThemeProvider>
        <ThemeProvider>
          <ThemeWrapper>
            <SelectWithDropdown {...props} />
          </ThemeWrapper>
        </ThemeProvider>
      </ChakraThemeProvider>,
    );

  it('should render and match the snapshot', () => {
    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });

  it('should display dropdown on click with appropriate length', () => {
    const { getByTestId } = renderComponent(props);
    fireEvent.click(getByTestId('icon-wrapper'));
    const element = getByTestId('items-wrapper');
    const list = element.querySelectorAll('button');
    expect(list.length).toEqual(LanguageItems.length);
  });

  it('should call action on select and hide list', () => {
    const handleOnSelect = jest.fn();
    const { getByTestId, getByText } = renderComponent({
      ...props,
      onSelect: handleOnSelect,
    });
    fireEvent.click(getByTestId('icon-wrapper'));
    fireEvent.click(getByText('English'));
    expect(handleOnSelect).toHaveBeenCalled();
    let element;
    try {
      element = getByTestId('items-wrapper');
    } catch (e) {
      expect(element).not.toBeDefined();
    }
  });
});
