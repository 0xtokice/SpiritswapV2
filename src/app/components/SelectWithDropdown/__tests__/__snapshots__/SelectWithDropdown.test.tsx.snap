// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectWithDropdown /> should render and match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0,
.c0 > div,
.c0 > svg,
.c0 > img {
  width: 20px;
  height: 20px;
}

<body
    class="chakra-ui-dark"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-hjcncd"
        data-testid="icon-wrapper"
      >
        <div
          class="c0"
        >
          <svg>
            language.svg
          </svg>
        </div>
        <label
          class="css-xmqerp"
        >
          undefined
        </label>
        <div
          class="c0"
        >
          <svg>
            caret-down.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-hjcncd"
      data-testid="icon-wrapper"
    >
      <div
        class="sc-bdvvtL kNCpAL"
      >
        <svg>
          language.svg
        </svg>
      </div>
      <label
        class="css-xmqerp"
      >
        undefined
      </label>
      <div
        class="sc-bdvvtL kNCpAL"
      >
        <svg>
          caret-down.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
